File: projects/JavaFXApps/ColorPuzzle/src/main/java/net/rickiekarp/colorpuzzlefx/ai/BruteForceSolver.java
package net.rickiekarp.colorpuzzlefx.ai;

import net.rickiekarp.colorpuzzlefx.core.Colors;
import net.rickiekarp.colorpuzzlefx.core.GameLogic;

public class BruteForceSolver implements Solver {
    private GameLogic game;

    private int indexLastColor = 0;

    @Override
    public void setGame(GameLogic game) {
        this.game = game;
    }

    @Override
    public Colors nextStep() {

        final Colors nextColor = Colors.values()[indexLastColor];

        if((indexLastColor +1) < Colors.values().length) {
            indexLastColor++;
        } else {
            indexLastColor = 0;
        }


        return nextColor;
    }
}

File: projects/JavaFXApps/ColorPuzzle/src/main/java/net/rickiekarp/colorpuzzlefx/ai/BruteForceSolver.kt
package net.rickiekarp.colorpuzzlefx.ai

import net.rickiekarp.colorpuzzlefx.core.Colors
import net.rickiekarp.colorpuzzlefx.core.GameLogic

class BruteForceSolver : Solver {
    private var game: GameLogic? = null
    private var indexLastColor = 0
    override fun setGame(game: GameLogic) {
        this.game = game
    }

    override fun nextStep(): Colors {
        val nextColor = Colors.values()[indexLastColor]
        if (indexLastColor + 1 < Colors.values().size) {
            indexLastColor++
        } else {
            indexLastColor = 0
        }
        return nextColor
    }
}